CLDR-13421 Add some alt=menu/variant names for ars, ckb

pedberg-icu opened the pull request:
<!--
Thank you for your pull request.
Please see http://cldr.unicode.org/index/process for general
information on contributing to CLDR.

You will be automatically asked to sign the contributors license before the PR is accepted.
- sign: https://cla-assistant.io/unicode-org/cldr
- license: http://www.unicode.org/copyright.html#License
-->

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13421
- [x] Updated PR title and link in previous line to include Issue number

The idea is to make available forms of the names for "ars" (Najdi Arabic) and "ckb" (Central/Sorani Kurdish) that make them sort with other related languages (e.g. Arabic, Kurdish); this is similar to what was done in [CLDR-11834](https://unicode-org.atlassian.net/browse/CLDR-11834) to have name variants that would sort Chinese languages together, e.g. `<language type="yue" alt="menu">Chinese, Cantonese</language>`

For "ars", if the current standard name is not already in a form like "Arabic, Najdi", then add an alt="menu" item that is in that form.

For "ckb" the choices are more complicated:
- If the current standard name is already in a form like "Kurdish, Sorani" then do nothing.
- If the current standard name is already in a form like "Sorani" or "Sorani Kurdish" then add an alt="menu" item that is like "Kurdish, Sorani".
- If the current standard name is in a form like "Kurdish, Central" then add add an alt="variant" (not menu) item that is like "Kurdish, Sorani".
- If the current standard name is in form like "Central Kurdish", then either
1. If we have localized data for the equivalent of both "Kurdish, Central" and "Kurdish, Sorani" then add the former as alt="menu" and the latter as alt="variant"
2. Otherwise, if we only have localized data for the equivalent of "Kurdish, Sorani", then add it as alt="menu" but mark as draft="contributed".

Many locales already have both names in a form like "Arabic, Najdi" and "Kurdish, Sorani": For example cs, el, es, id, it, ms, sk. No change was made for these. Other locales already have one of the names in that form, so only the other was changed.

When adding names, the formatting (use of commas, dashes or parens for the modifier part) was modeled on the way the loicale already handled other such names with modifiers (including if it already had one of the names for ars or ckb in the desired form)

This was only checked/modified for the ~40 locales for which Apple had in-house localization resources to provied the updated names.

macchiati commented:
> I don't think alt="menu" show up in the Survey tool, but alt="variant" would.
All of the alt values should show up; if they don't, then that's a bug. Could you check?

pedberg-icu commented:
> * Could you check the spelling for Sorani in Finnish?  Why are they different between default and the alt="menu"? soran**i** vs. kurdi – soran**î**

The correct spelling seems to be soran**î** (per our linguists and https://fi.wikipedia.org/wiki/Soranî) so that is the form I added. I did not want to change the default version without discussion.

> * Why do some language have the draft status and most don't?
>   <language type="ckb" alt="menu" **draft="contributed"**>庫德文（索拉尼）

That is discussed in my overview comments, see #2 after the last bullet.

> * in En.xml for ckb a new variant name is added in En: Kurdish, Sorani. I think this should be raised for visibility. I don't think alt="menu" show up in the Survey tool, but alt="variant" would. So, there would be impact to other languages. Should agree on this in the TC.

I think both the alt="menu" and alt="variant" names for languages will show up in Survey Tool if the coverage includes the language name at all (ckb is at modern coverage). I agree we should discuss in TC, I have added to agenda for 2019-Dec-04. We can adjust the behavior here with a new PR per the results of that discussion.

pedberg-icu commented:
I have filed JIRA tickets for two follow-on issues:
- The issue about correct Finnish spelling "sorani" or "soranî" is covered by [CLDR-13464](https://unicode-org.atlassian.net/browse/CLDR-13464)
- The issue about whether to make the alt="menu" and alt="variant" forms for "ckb" show up in Survey Tool for languages that do not already have those variants is covered by [CLDR-13463](https://unicode-org.atlassian.net/browse/CLDR-13463)
